File: AtCoder/arc114.py
N = int(input())
X = list(map(int, input().split()))
X.sort()
judge = [True]*N
S =[2,3,5,7,11,13,17,19,23,29,31,37,41,43,47]
ans = 1

for s in S:
  judge_2 = True
  if s in X:
    ans *= s
    judge_2 = False
  for x in range(0, N):
    if X[x] % s == 0 and judge[x]:
      if judge_2:
        ans *= s
        s = 1
      judge[x] = False


print(ans)

N = int(input())
X = list(map(int, input().split()))
X.sort()
judge = [True]*N
S =[2,3,5,7,11,13,17,19,23,29,31,37,41,43,47]
ans = 1

for s in S:
  cnt = 0
  for x in range(0, N):
    if X[x]%s==0 and judge[x]:
      cnt += 1
      judge[x] = False
  if cnt > 0:
    ans *= s
print(ans)

File: AtCoder/arc114.rb
# ARC114
# A
require 'prime'
N = gets.to_i
X = gets.split.map &:to_i

# 50までの数字のうち素数のみをprimeに格納（全部で15コ）
prime = []
Prime.each(50){|i| prime<<i}

# 上記15コの素数のうち、掛け合わせて作れるすべての値をpprimeに格納
pprime = []
1.upto(15) do |i|
  combi = prime.combination(i).to_a
  combi.each do |j|
    pprime << j.inject(:*)
  end
end
pprime.sort!

# pprimeに格納されている整数のうち、Xすべてと最小公約数が1にならない最小値がanswer
ans = 0
pprime.each do |i|
  if X.all?{|e| e.gcd(i) != 1}
    ans = i
    break
  end
end

puts ans

